src/routes/concepts/components/event-handlers.mdx

```diff
@@ -3,68 +3,199 @@ title: Event Handlers
 
 ```tsx
-const handler = (data, event) => {
-	console.log("Data:", data, "Event:", event);
-};
+<div onMouseMove={handleCustomEvent} />
+```
 
-<button onClick={[handler, "Hello!"]}>Click Me</button>;
+<Callout type="tip" title="Ocassional Events">
+
+Rather than using delegated events for events that happen infrequently, **native events** are a better solution.
+Since these events happen in specific circumstances, they do not benefit from the performance improvements of event delegation.
+
+```tsx
+<div on:mousemove={handleCustomEvent} />
+```
+
+</Callout>
+
+- `event.stopPropogation()` does not work as expected since events are attached to the `document` rather than the `element`.
+
+With cases like this, a native event is recommended.
+As an example, using a native event would stop the following event from reaching the `div native` handler, which is _not_ the case for delegated events.
+You can [view this example in the Solid Playground](https://playground.solidjs.com/anonymous/c5346f84-01e4-4080-8ace-4443ffd0bb10).
+
+```tsx
+onMount(() => {
+	ref.addEventListener("click", () => {
+		console.log("div native");
+	});
+});
+<div ref={ref}>
+	<button
+		onClick={(event) => {
+			event.stopPropagation();
+			console.log("button");
+		}}
+	>
+		button
+	</button>
+</div>;
+```
+
+```shellsession title="Console output"
+// Button clicked
+div native
+button
+```
+
+You can solve this by switching the `button` event to using a native event:
+
+```tsx ins="on:click"
+// ...
+<button
+	on:click={(event) => {
+		event.stopPropagation();
+		console.log("button");
+	}}
+>
+	button
+</button>
+// ...
+```
+
+```shellsession title="Console output"
+// Button clicked
+button
 ```
 
-In this example, the string `'Hello!'` (the second item in the array) is passed as the data parameter (the first argument of the handler function) when the button is clicked.
+[See how this solution differs in the Solid Playground](https://playground.solidjs.com/anonymous/9e2deddc-2e83-4ac2-8ee0-49c7c3a45d11).
 
-By binding events in this way, the overhead of using JavaScript's [bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) method is avoided.
+- Portals propagate events following the _component tree_ and not the _DOM tree_, making them easier to use.
+  This means when a `Portal` gets attached to the `body`, any events will propagate up to the `container`.
+
+```tsx
+<div class="container" onInput={() => console.log("portal key press")}>
+	<Portal mount={document.body}>
+		<input onInput={() => console.log("input key press")} />
+	</Portal>
+</div>
+```
+
+<Callout type="info" title="onInput / onChange">
+
+    `onChange` and `onInput` events work according to their native behavior:
+    - `onInput` will fire immediately after the value has changed
+    - In `<input>` fields, `onChange` will only fire after the field loses focus.
+
+</Callout>
 
 ## Delegated Events
```

Comment:
@titoBouzout @LadyBluenotes shouldn't this be a subheader under "Event delegation"?

```suggestion
### Delegated Events
```

Reply:
Also maybe switch to "List of delegated events"?